ega-data-api-res/src/main/java/eu/elixir/ega/ebi/reencryptionmvc/service/internal/CacheResServiceImpl.java

```diff
@@ -284,7 +287,9 @@ else if (sourceFormat.equalsIgnoreCase("symmetricgpg")) {
             throw new GeneralStreamingException(sessionId + " Error Location: " + errorLocation + "\n" + ex.toString(), 10);
         } finally {
             try {
-                in.close();
+                if (in != null) {
```

Comment:
Cool!

Don't the other resources (encryptedDigestOut, eOut) need a similar `if (resource != null)` wrapper?
Or maybe they all could be wrapped in a try-with-resources block?

Reply:
While debugging I didn't get the other resources null, but for the safe side added it.